roles/README.adoc

```diff
@@ -19,6 +19,75 @@ The role internally would have the logic to decide whether to use ntpd, chronyd,
 However, when the underlying implementations become too divergent, for example implementing an email server with postfix or sendmail, then 
 separate roles are encouraged.
 
+Design roles to accomplish a specific, guaranteed outcome and limit the scope of the role to that outcome.
+This will help abstracting differences between different providers (see above), and help the user to focus on the functionality, not on technical details.
+====
+
+* Design the interface focused on the functionality, not on the software implementation behind it.
++
+[%collapsible]                                                                                                        
+====
+Explanations:: Limit the consumer's need to understand specific implementation details about a collection, the role names within, and the file names. Presenting the collection as a low-code/no-code "automation application" provides the developer flexibility as the content grows and matures, and limits change the consumer may have to make in a later version.
+
+Examples::
+* In the first example, the `mycollection.run` role has been designed to be an entry-point for the execution of multiple roles.  `mycollection.run` provides a standard interface for the user, without exposing the details of the underlying automations.  The implementation details of `thing_1` and `thing_2` may be changed by the developer without impacting the user as long as the interface of `mycollection.run` does not change.
+
+.Do this:
+[source,yaml]
+----
+
+- hosts: all
+  gather_facts: false
+  tasks:
+  - name: Perform several actions
+    include_role: mycollection.run
+    vars:
+      actions:
+      - name: thing_1
+        vars:
+          thing_1_var_1: 1
+          thing_1_var_2: 2
+      - name: thing_2
+        vars:
+          thing_2_var_1: 1
+          thing_2_var_2: 2
+
+----
+
+* In this example, the user must maintain awareness of the structure of the `thing_1` and `thing_2` roles, and the order of operations necessary for these roles to be used together. If the implementation of these roles is changed, the user will need to modify their playbook.
+.Don't do this:
+[source,yaml]
+----
+- hosts: all
+  gather_facts: false
+  tasks:
+  - name: Do thing_1
+    include_role: 
+      name: mycollection.thing_1
+      tasks_from: thing_1.yaml
+      vars:
+        thing_1_var_1: 1
+        thing_1_var_2: 2
+  - name: Do thing_2
+    include_role: 
+      name: mycollection.thing_2
+      tasks_from: thing_2.yaml
+      vars:
+        thing_2_var_1: 1
+        thing_2_var_2: 2
+----
+
+====
+
+* Place content common to multiple roles in a single reusable role, "common" is a typical name for this role when roles are packaged in a collection.
+* Author loosely coupled, hierarchical content.
```

Comment:
```suggestion
  Author loosely coupled, hierarchical content.
```
Again, two phrases with the same explanation and rationale, so merge into one bullet.